nuxeo-notification-stream-core/src/main/java/org/nuxeo/ecm/platform/notification/processor/NotificationProcessor.java

```diff
@@ -17,14 +16,21 @@
  *      Nuxeo
  */
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+package org.nuxeo.ecm.platform.notification.processor;
 
-public class LogDispatcher extends Dispatcher {
-    private static final Log log = LogFactory.getLog(LogDispatcher.class);
+import java.util.Map;
 
+import org.nuxeo.ecm.platform.notification.NotificationStreamConfig;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ * @since XXX
+ */
+public class NotificationProcessor implements StreamProcessorTopology {
     @Override
-    public void dispatch() {
-        log.info("Dispatched");
+    public Topology getTopology(Map<String, String> options) {
+        return Framework.getService(NotificationStreamConfig.class).buildTopology(options);
```

Comment:
why not having NotificationComponent implements StreamProcessorTopology so you can remove buildTopology from the NotificationStreamConfig interface ?

Reply:
:+1: